### PR TITLE
Added author migrate command

### PR DESCRIPTION
**Pull request by jmbarne3**
**Description**
Adds the author migration command. This takes a required json file as the first parameter, and has the optional parameters to provide a default author (true by default) and to specify that author (set to 1 by default). Documentation will be added to the readme and wiki afterwards.

**Motivation and Context**
Reasons.

**How Has This Been Tested?**
This migration has successfully run on my local environment. Needs to be deployed to dev and tested.

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

- [x] My code follows the code style of this project.
- [x] My change requires an update to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jmbarne3** [comment]
Wiki is up in a super simplified form: https://github.com/UCF/WP-Author-Migration/wiki

File: includes/class-wpam-author.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		 * Maps incoming user data to an existing user account.
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 * @return bool True if the user is mapped, false if it is not mapped.
+		 */
+		private function map_existing_user() {
+			$user = null;
+
+			$user = get_user_by( 'id', $this->old_id );
+
```
Comment (cjg89) [inline includes/class-wpam-author.php]: There is no `if ( $user ) { ... }` check after attempting to fetch the user by ID.  Without a check that returns early when the user has an `old_id`, `$user` will immediately get overridden by line 48

File: includes/class-wpam-author-migrate.php [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+		}
+
+		/**
+		 * Runs the migration steps
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 */
+		public function migrate() {
+			$query = new WP_Query( array(
+				'post_type'      => 'post',
```
[the remark this answers is not in the document]
Comment (jmbarne3) [reply includes/class-wpam-author-migrate.php]: Planning on implementing other post type options. Just wanted to concentrate on getting it working for posts on the first iteration.

File: includes/class-wpam-author.php (hunk 1 above)
Comment (jmbarne3) [reply includes/class-wpam-author.php]: Updated in 07a5a67.

File: includes/class-wpam-author-migrate.php [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+					$author = $this->default_author;
+				} else {
+					$this->cannot_update++;
+					return;
+				}
+			}
+
+			global $wpdb;
+
+			// If the author
```
[the remark this answers is not in the document]
Comment (jmbarne3) [reply includes/class-wpam-author-migrate.php]: Updated in a805423

**Pull request merged**